dds/idl/marshal_generator.cpp

```diff
@@ -2777,6 +2777,10 @@ namespace {
       be_global->impl_ <<
         "  const Encoding& encoding = strm.encoding();\n"
         "  ACE_UNUSED_ARG(encoding);\n";
+      if (is_appendable) {
+        be_global->impl_ <<
+          "  bool end_of_stream = false;\n";
```

Comment:
This name could be misleading since the struct is not the whole stream.

```suggestion
          "  bool reached_end_of_struct = false;\n";
```